tests/ltp/patches/ltp_listxattr_listxattr01_fix.patch

```diff
@@ -0,0 +1,49 @@
++ Patch Description: Tests were failing as no xattr support in loop device
++ So modofied the tests to use root filesystem with xattr support
+diff --git a/testcases/kernel/syscalls/listxattr/listxattr01.c b/testcases/kernel/syscalls/listxattr/listxattr01.c
+index 62198b2a7..ca3e70232 100644
+--- a/testcases/kernel/syscalls/listxattr/listxattr01.c
++++ b/testcases/kernel/syscalls/listxattr/listxattr01.c
+@@ -31,7 +31,9 @@
+ #define VALUE  "test"
+ #define VALUE_SIZE     (sizeof(VALUE) - 1)
+ #define KEY_SIZE    (sizeof(SECURITY_KEY1) - 1)
+-#define TESTFILE    "testfile"
++#define TESTFILE    "mntpoint/testfile"
++#define MNTPOINT    "mntpoint"
++#define DIR_MODE    (S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH)
+
+ static int has_attribute(const char *list, int llen, const char *attr)
+ {
+@@ -64,17 +66,29 @@ static void verify_listxattr(void)
+ }
+
+ static void setup(void)
+-{
++{
++       rmdir(MNTPOINT);
++       SAFE_MKDIR(MNTPOINT, DIR_MODE);
++       SAFE_MOUNT("/dev/vda", MNTPOINT, "ext4", 0, NULL);
++
+        SAFE_TOUCH(TESTFILE, 0644, NULL);
+
+        SAFE_SETXATTR(TESTFILE, SECURITY_KEY1, VALUE, VALUE_SIZE, XATTR_CREATE);
+ }
+
++static void cleanup(void)
++{
++        remove(TESTFILE);
++        SAFE_UMOUNT(MNTPOINT);
```

Comment:
What is being unmounted here?

Reply:
Folder ./mountpoint is created
/dev/vda is mounted to ./mountpoint folder
test runs and then
 It is unmounted
mountpoint folder is removed.

Reply:
The root file system shouldn't be unmounted here.

Reply:
@prp  here we are unmounting /dev/vda. Do you mean this shoulnd't be unmounted?

Reply:
Indeed. It doesn't make sense for a userspace process to try to unmount the root file system while running.

Reply:
@prp what about mounting? You said userspace process shouldn't unmount, but it is mounting at the beginning of the test. In the test to leave the image clean, we are unmounting it and deleting the mounterd folder. Actually after PR https://github.com/lsds/sgx-lkl/pull/562 merged, we will have fresh image file for each LTP test and then we can remove the unmount code here.

Reply:
See my comment above -- it makes no sense to mount `/dev/vda`. This is the root partition, and it has already been mounted by SGX-LKL during start-up.

Reply:
> Indeed. It doesn't make sense for a userspace process to try to unmount the root file system while running.

@prp, in setup function of the test case "/dev/vda" is mounted to "./mnt_point". This code is counter part of un-mounting "./mnt_point". we will remove this un-mount code. 
As I mentioned in above discussion, the test case is passed by creating file in "/" and "/ltp".  As per initial investigation the root file system is not mounted with "xattr" option, I am further checking on why setxattr is passed on this file.

Reply:
As I said several times now, mounting `/dev/vda` makes no sense.

If the root file system doesn't support `xattr`, this test cannot be run, and it is indeed surprising why it would pass.